syntax="proto3";
package market_communication_with_client;
service MarketPlace
{
    rpc SearchItem(SearchRequest) returns (stream ProductDisplayResponse);
    rpc BuyItem(BuyRequest) returns (StatusResponse);
    rpc AddWish(WishRequest) returns (StatusResponse);
    rpc RateItem(RateRequest) returns (StatusResponse);
}
service BuyerNotificationServer
{
    rpc ReceiveNotification(ProductDisplayResponse) returns (StatusResponse);
}
enum Category
{
    ELECTRONICS=0;
    FASHION=1;
    OTHERS=2;
    ANY=3;
}
message SearchRequest
{
    string item_name=1;
    Category category=2;
}
message ProductDisplayResponse
{
    uint64 id = 1;
    float price = 2;
    string name = 3;
    Category productCategory=4;
    uint32 quantityRemaining=5;
    float rating = 6;
    string sellerAddress=7;
    string description = 8;
}
message BuyRequest
{
    uint64 id=1;
    uint32 quantity=2;
}
message WishRequest
{
    string uuid = 1;
    uint64 id= 2;
}
message RateRequest
{
    uint64 id=1;
    int32 rating=2;
}
message StatusResponse
{
    string status=1;
}
